### ISSUE TITLE
[Bug]: The error log reported by indexcoord is inaccurate.

### ISSUE THREAD

**Issue by JackLCL** [milvus-io/milvus#16771]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version:2.0.2
- Deployment mode(standalone or cluster):cluster
- SDK version(e.g. pymilvus v2.0.0rc2):java sdk 2.0.4
- OS(Ubuntu or CentOS): N/A
- CPU/Memory: N/A
- GPU: None
- Others: N/A
```

### Current Behavior

The disconnection between the indexnode and etcd causes the index to fail to be created normally, but the indexcoord reports that the memory of the indexnode is insufficient.

### Expected Behavior

Indexcoord can report the error that the indexnode is disconnected from etcd normally.

### Steps To Reproduce

```markdown
1. start milvus cluster
2. use sata disk for etcd
3. insert 10w 768d vectors
4. create index
```

### Milvus Log

```markdown
indexcoord log:
[2022/05/01 21:29:13.185 +08:00] [INFO] [grpclog.go:37] ["[core]Channel Connectivity change to TRANSIENT_FAILURE"]
[2022/05/01 21:29:13.186 +08:00] [INFO] [grpclog.go:37] ["[core]Subchannel Connectivity change to CONNECTING"]
[2022/05/01 21:29:13.186 +08:00] [INFO] [grpclog.go:37] ["[core]Subchannel Connectivity change to CONNECTING"]
[2022/05/01 21:29:13.186 +08:00] [INFO] [grpclog.go:37] ["[transport]transport: loopyWriter.run returning. connection error: desc = \"transport is closing\""]
[2022/05/01 21:29:13.186 +08:00] [INFO] [grpclog.go:37] ["[transport]transport: loopyWriter.run returning. connection error: desc = \"transport is closing\""]
[2022/05/01 21:29:13.186 +08:00] [INFO] [grpclog.go:37] ["[core]pickfirstBalancer: UpdateSubConnState: 0xc00060af20, {CONNECTING <nil>}"]
[2022/05/01 21:29:13.186 +08:00] [INFO] [grpclog.go:37] ["[core]Channel Connectivity change to CONNECTING"]
[2022/05/01 21:29:13.186 +08:00] [INFO] [grpclog.go:37] ["[core]pickfirstBalancer: UpdateSubConnState: 0xc000c48900, {CONNECTING <nil>}"]
[2022/05/01 21:29:13.186 +08:00] [INFO] [grpclog.go:37] ["[core]Subchannel picks a new address \"10.245.18.9:21121\" to connect"]
[2022/05/01 21:29:13.186 +08:00] [INFO] [grpclog.go:37] ["[core]Channel Connectivity change to CONNECTING"]
[2022/05/01 21:29:13.186 +08:00] [INFO] [grpclog.go:37] ["[core]Subchannel picks a new address \"10.245.18.9:21121\" to connect"]
[2022/05/01 21:29:13.186 +08:00] [WARN] [grpclog.go:46] ["[core]grpc: addrConn.createTransport failed to connect to {10.245.18.9:21121 10.245.18.9:21121 <nil> 0 <nil>}. Err: connection error: desc = \"transport: Error while dialing dial tcp 10.245.18.9:21121: connect: connection refused\". Reconnecting..."]
[2022/05/01 21:29:13.186 +08:00] [INFO] [grpclog.go:37] ["[core]Subchannel Connectivity change to TRANSIENT_FAILURE"]
[2022/05/01 21:29:13.186 +08:00] [INFO] [grpclog.go:37] ["[core]pickfirstBalancer: UpdateSubConnState: 0xc00060af20, {TRANSIENT_FAILURE connection error: desc = \"transport: Error while dialing dial tcp 10.245.18.9:21121: connect: connection refused\"}"]
[2022/05/01 21:29:13.186 +08:00] [INFO] [grpclog.go:37] ["[core]Channel Connectivity change to TRANSIENT_FAILURE"]
[2022/05/01 21:29:13.187 +08:00] [WARN] [grpclog.go:46] ["[core]grpc: addrConn.createTransport failed to connect to {10.245.18.9:21121 10.245.18.9:21121 <nil> 0 <nil>}. Err: connection error: desc = \"transport: Error while dialing dial tcp 10.245.18.9:21121: connect: connection refused\". Reconnecting..."]
[2022/05/01 21:29:13.187 +08:00] [INFO] [grpclog.go:37] ["[core]Subchannel Connectivity change to TRANSIENT_FAILURE"]
[2022/05/01 21:29:13.187 +08:00] [INFO] [grpclog.go:37] ["[core]pickfirstBalancer: UpdateSubConnState: 0xc000c48900, {TRANSIENT_FAILURE connection error: desc = \"transport: Error while dialing dial tcp 10.245.18.9:21121: connect: connection refused\"}"]
[2022/05/01 21:29:13.187 +08:00] [INFO] [grpclog.go:37] ["[core]Channel Connectivity change to TRANSIENT_FAILURE"]
[2022/05/01 21:29:13.204 +08:00] [WARN] [etcd_kv.go:648] ["Slow etcd operation load with prefix2"] ["time spent"=9.131180981s]
[2022/05/01 21:29:13.204 +08:00] [DEBUG] [meta_table.go:120] ["IndexCoord reloadMeta mt.client.LoadWithPrefix2"] [indexBuildID=432665387088216066] []
[2022/05/01 21:29:13.232 +08:00] [INFO] [grpclog.go:37] ["[transport]transport: loopyWriter.run returning. connection error: desc = \"transport is closing\""]
[2022/05/01 21:29:13.243 +08:00] [DEBUG] [meta_table.go:104] ["IndexCoord metaTable saveIndexMeta "] [key=indexes/432665387088216066] []
[2022/05/01 21:29:13.243 +08:00] [DEBUG] [meta_table.go:110] ["IndexCoord metaTable saveIndexMeta success"] [meta.revision=73284]
[2022/05/01 21:29:13.243 +08:00] [DEBUG] [index_coord.go:898] ["The version of the task has been updated"] [indexBuildID=432665387088216066]
[2022/05/01 21:29:13.243 +08:00] [DEBUG] [node_manager.go:106] ["IndexCoord NodeManager PeekClient"]
[2022/05/01 21:29:13.243 +08:00] [DEBUG] [node_manager.go:118] ["IndexCoord peek IndexNode client from pq"] ["data size"=2397014016]
[2022/05/01 21:29:13.243 +08:00] [ERROR] [node_manager.go:121] ["No IndexNode available"] ["data size"=2397014016] ["IndexNode must have memory size"=14382084096] [stack="github.com/milvus-io/milvus/internal/indexcoord.(*NodeManager).PeekClient\n\t/go/src/github.com/milvus-io/milvus/internal/indexcoord/node_manager.go:121\ngithub.com/milvus-io/milvus/internal/indexcoord.(*IndexCoord).assignTaskLoop\n\t/go/src/github.com/milvus-io/milvus/internal/indexcoord/index_coord.go:900"]
[2022/05/01 21:29:13.243 +08:00] [ERROR] [node_manager.go:128] ["IndexCoord NodeManager PeekClient"] ["There is no IndexNode client corresponding to NodeID"=-1] [stack="github.com/milvus-io/milvus/internal/indexcoord.(*NodeManager).PeekClient\n\t/go/src/github.com/milvus-io/milvus/internal/indexcoord/node_manager.go:128\ngithub.com/milvus-io/milvus/internal/indexcoord.(*IndexCoord).assignTaskLoop\n\t/go/src/github.com/milvus-io/milvus/internal/indexcoord/index_coord.go:900"]
[2022/05/01 21:29:13.243 +08:00] [WARN] [index_coord.go:902] ["IndexCoord assignmentTasksLoop can not find available IndexNode"]
[2022/05/01 21:29:13.243 +08:00] [DEBUG] [meta_table.go:506] ["IndexCoord metaTable LoadMetaFromETCD"] [indexBuildID=432665387088216072] [revision=73284] [ok=true]
[2022/05/01 21:29:13.243 +08:00] [DEBUG] [meta_table.go:509] ["IndexCoord metaTable LoadMetaFromETCD"] [meta.revision=73284] [revision=73284]
[2022/05/01 21:29:13.243 +08:00] [DEBUG] [index_coord.go:827] ["IndexCoord watchMetaLoop PUT"] [IndexBuildID=432665387088216072] [reload=false]
[2022/05/01 21:29:13.243 +08:00] [DEBUG] [index_coord.go:816] ["IndexCoord watchMetaLoop find meta updated."]
[2022/05/01 21:29:13.244 +08:00] [DEBUG] [index_coord.go:890] ["IndexCoord find unassigned tasks "] ["Unassigned tasks number"=4] ["Available IndexNode IDs"="[276,268,271,299,307,272]"]
[2022/05/01 21:29:13.244 +08:00] [DEBUG] [meta_table.go:217] ["IndexCoord metaTable update UpdateVersion"] [IndexBuildId=432665387088216070]
[2022/05/01 21:29:13.244 +08:00] [DEBUG] [meta_table.go:229] ["IndexCoord metaTable update UpdateVersion"] [IndexBuildId=432665387088216070] [Version=73282]
[2022/05/01 21:29:13.244 +08:00] [DEBUG] [index_coord.go:822] ["IndexCoord watchMetaLoop"] [event.Key="YnktZGV2L21ldGEvaW5kZXhlcy80MzI2NjUzODcwODgyMTYwNjY="] [event.V="indexBuildID:432665387088216066 state:Unissued req:<indexBuildID:432665387088216066 index_name:\"_default_idx\" indexID:432676578062761985 data_paths:\"file/insert_log/432676572701917185/432676572701917186/432676606793482241/102/432676607619497987\" data_paths:\"file/insert_log/432676572701917185/432676572701917186/432676606793482241/102/432676609624899592\" data_paths:\"file/insert_log/432676572701917185/432676572701917186/432676606793482241/102/432676610987786243\" data_paths:\"file/insert_log/432676572701917185/432676572701917186/432676606793482241/102/432676612232970248\" data_paths:\"file/insert_log/432676572701917185/432676572701917186/432676606793482241/102/432676613569904643\" data_paths:\"file/insert_log/432676572701917185/432676572701917186/432676606793482241/102/432676614906839048\" data_paths:\"file/insert_log/432676572701917185/432676572701917186/432676606793482241/102/432676616217559043\" data_paths:\"file/insert_log/432676572701917185/432676572701917186/432676606793482241/102/432676617318825987\" data_paths:\"file/insert_log/432676572701917185/432676572701917186/432676606793482241/102/432676618891427843\" data_paths:\"file/insert_log/432676572701917185/432676572701917186/432676606793482241/102/432676620149719048\" data_paths:\"file/insert_log/432676572701917185/432676572701917186/432676606793482241/102/432676621381795843\" data_paths:\"file/insert_log/432676572701917185/432676572701917186/432676606793482241/102/432676622784528387\" data_paths:\"file/insert_log/432676572701917185/432676572701917186/432676606793482241/102/432676624055664643\" data_paths:\"file/insert_log/432676572701917185/432676572701917186/432676606793482241/102/432676625366646787\" data_paths:\"file/insert_log/432676572701917185/432676572701917186/432676606793482241/102/432676626781962243\" data_paths:\"file/insert_log/432676572701917185/432676572701917186/432676606793482241/102/432676628027408387\" data_paths:\"file/insert_log/432676572701917185/432676572701917186/432676606793482241/102/432676629272592392\" data_paths:\"file/insert_log/432676572701917185/432676572701917186/432676606793482241/102/432676630426025992\" data_paths:\"file/insert_log/432676572701917185/432676572701917186/432676606793482241/102/432676631788912643\" data_paths:\"file/insert_log/432676572701917185/432676572701917186/432676606793482241/102/432676633309347843\" data_paths:\"file/insert_log/432676572701917185/432676572701917186/432676606793482241/102/432676634344816643\" data_paths:\"file/insert_log/432676572701917185/432676572701917186/432676606793482241/102/432676635668905987\" data_paths:\"file/insert_log/432676572701917185/432676572701917186/432676606793482241/102/432676636822339587\" data_paths:\"file/insert_log/432676572701917185/432676572701917186/432676606793482241/102/432676638303191043\" data_paths:\"file/insert_log/432676572701917185/432676572701917186/432676606793482241/102/432676639718768643\" data_paths:\"file/insert_log/432676572701917185/432676572701917186/432676606793482241/102/432676640767344648\" data_paths:\"file/insert_log/432676572701917185/432676572701917186/432676606793482241/102/432676642340470787\" data_paths:\"file/insert_log/432676572701917185/432676572701917186/432676606793482241/102/432676643900227587\" data_paths:\"file/insert_log/432676572701917185/432676572701917186/432676606793482241/102/432676645263114243\" data_paths:\"file/insert_log/432676572701917185/432676572701917186/432676606793482241/102/432676646744227843\" data_paths:\"file/insert_log/432676572701917185/432676572701917186/432676606793482241/102/432676648238710787\" data_paths:\"file/insert_log/432676572701917185/432676572701917186/432676606793482241/102/432676649575645187\" data_paths:\"file/insert_log/432676572701917185/432676572701917186/432676606793482241/102/432676650886103048\" data_paths:\"file/insert_log/432676572701917185/432676572701917186/432676606793482241/102/432676652157501443\" data_paths:\"file/insert_log/432676572701917185/432676572701917186/432676606793482241/102/432676653900759043\" data_paths:\"file/insert_log/432676572701917185/432676572701917186/432676606793482241/102/432676654818263043\" data_paths:\"file/insert_log/432676572701917185/432676572701917186/432676606793482241/102/432676656220995587\" data_paths:\"file/insert_log/432676572701917185/432676572701917186/432676606793482241/102/432676657600135171\" data_paths:\"file/insert_log/432676572701917185/432676572701917186/432676606793482241/102/432676659012567048\" data_paths:\"file/insert_log/432676572701917185/432676572701917186/432676606793482241/102/432676660323287048\" data_paths:\"file/insert_log/432676572701917185/432676572701917186/432676606793482241/102/432676661530460163\" data_paths:\"file/insert_log/432676572701917185/432676572701917186/432676606793482241/102/432676662879191048\" data_paths:\"file/insert_log/432676572701917185/432676572701917186/432676606793482241/102/432676664229494787\" data_paths:\"file/insert_log/432676572701917185/432676572701917186/432676606793482241/102/432676665264701443\" data_paths:\"file/insert_log/432676572701917185/432676572701917186/432676606793482241/102/432676666522992643\" data_paths:\"file/insert_log/432676572701917185/432676572701917186/432676606793482241/102/432676668056797192\" data_paths:\"file/insert_log/432676572701917185/432676572701917186/432676606793482241/102/432676669419683848\" data_paths:\"file/insert_log/432676572701917185/432676572701917186/432676606793482241/102/432676670795939843\" data_paths:\"file/insert_log/432676572701917185/432676572701917186/432676606793482241/102/432676672041385987\" data_paths:\"file/insert_log/432676572701917185/432676572701917186/432676606793482241/102/432676673456701443\" data_paths:\"file/insert_log/432676572701917185/432676572701917186/432676606793482241/102/432676675081994243\" data_paths:\"file/insert_log/432676572701917185/432676572701917186/432676606793482241/102/432676676576215043\" data_paths:\"file/insert_log/432676572701917185/432676572701917186/432676606793482241/102/432676677703434243\" data_paths:\"file/insert_log/432676572701917185/432676572701917186/432676606793482241/102/432676678856867843\" data_paths:\"file/insert_log/432676572701917185/432676572701917186/432676606793482241/102/432676680167587843\" data_paths:\"file/insert_log/432676572701917185/432676572701917186/432676606793482241/102/432676681792880648\" data_paths:\"file/insert_log/432676572701917185/432676572701917186/432676606793482241/102/432676683103600643\" data_paths:\"file/insert_log/432676572701917185/432676572701917186/432676606793482241/102/432676683850973192\" data_paths:\"file/insert_log/432676572701917185/432676572701917186/432676606793482241/102/432676686681866243\" data_paths:\"file/insert_log/432676572701917185/432676572701917186/432676606793482241/102/432676687966371843\" data_paths:\"file/insert_log/432676572701917185/432676572701917186/432676606793482241/102/432676689119805443\" data_paths:\"file/insert_log/432676572701917185/432676572701917186/432676606793482241/102/432676690482954243\" data_paths:\"file/insert_log/432676572701917185/432676572701917186/432676606793482241/102/432676691728138243\" data_paths:\"file/insert_log/432676572701917185/432676572701917186/432676606793482241/102/432676692802928643\" data_paths:\"file/insert_log/432676572701917185/432676572701917186/432676606793482241/102/432676694126755843\" data_paths:\"file/insert_log/432676572701917185/432676572701917186/432676606793482241/102/432676695332880387\" data_paths:\"file/insert_log/432676572701917185/432676572701917186/432676606793482241/102/432676696577802248\" data_paths:\"file/insert_log/432676572701917185/432676572701917186/432676606793482241/102/432676697836093443\" data_paths:\"file/insert_log/432676572701917185/432676572701917186/432676606793482241/102/432676698950467592\" data_paths:\"file/insert_log/432676572701917185/432676572701917186/432676606793482241/102/432676700300247043\" data_paths:\"file/insert_log/432676572701917185/432676572701917186/432676606793482241/102/432676701585014787\" data_paths:\"file/insert_log/432676572701917185/432676572701917186/432676606793482241/102/432676703144771587\" data_paths:\"file/insert_log/432676572701917185/432676572701917186/432676606793482241/102/432676704297943043\" data_paths:\"file/insert_log/432676572701917185/432676572701917186/432676606793482241/102/432676705844592643\" data_paths:\"file/insert_log/432676572701917185/432676572701917186/432676606793482241/102/432676707155312643\" data_paths:\"file/insert_log/432676572701917185/432676572701917186/432676606793482241/102/432676708295901187\" data_paths:\"file/insert_log/432676572701917185/432676572701917186/432676606793482241/102/432676709750800387\" data_paths:\"file/insert_log/432676572701917185/432676572701917186/432676606793482241/102/432676711231913987\" type_params:<key:\"dim\" value:\"768\" > index_params:<key:\"metric_type\" value:\"IP\" > index_params:<key:\"params\" value:\"{\\\"nlist\\\":2048}\" > index_params:<key:\"index_type\" value:\"IVF_FLAT\" > num_rows:780278 field_schema:<fieldID:102 name:\"data\" data_type:FloatVector type_params:<key:\"dim\" value:\"768\" > > > mark_deleted:true version:73282 "] [IndexBuildID=432665387088216066] []
[2022/05/01 21:29:13.253 +08:00] [DEBUG] [meta_table.go:104] ["IndexCoord metaTable saveIndexMeta "] [key=indexes/432665387088216070] []
[2022/05/01 21:29:13.253 +08:00] [DEBUG] [meta_table.go:110] ["IndexCoord metaTable saveIndexMeta success"] [meta.revision=73284]
[2022/05/01 21:29:13.253 +08:00] [DEBUG] [index_coord.go:898] ["The version of the task has been updated"] [indexBuildID=432665387088216070]
[2022/05/01 21:29:13.253 +08:00] [DEBUG] [node_manager.go:106] ["IndexCoord NodeManager PeekClient"]
[2022/05/01 21:29:13.253 +08:00] [DEBUG] [node_manager.go:118] ["IndexCoord peek IndexNode client from pq"] ["data size"=2394132480]
[2022/05/01 21:29:13.253 +08:00] [ERROR] [node_manager.go:121] ["No IndexNode available"] ["data size"=2394132480] ["IndexNode must have memory size"=14364794880] [stack="github.com/milvus-io/milvus/internal/indexcoord.(*NodeManager).PeekClient\n\t/go/src/github.com/milvus-io/milvus/internal/indexcoord/node_manager.go:121\ngithub.com/milvus-io/milvus/internal/indexcoord.(*IndexCoord).assignTaskLoop\n\t/go/src/github.com/milvus-io/milvus/internal/indexcoord/index_coord.go:900"]
[2022/05/01 21:29:13.253 +08:00] [DEBUG] [meta_table.go:506] ["IndexCoord metaTable LoadMetaFromETCD"] [indexBuildID=432665387088216066] [revision=73284] [ok=true]
[2022/05/01 21:29:13.253 +08:00] [DEBUG] [meta_table.go:509] ["IndexCoord metaTable LoadMetaFromETCD"] [meta.revision=73284] [revision=73284]
[2022/05/01 21:29:13.253 +08:00] [DEBUG] [index_coord.go:827] ["IndexCoord watchMetaLoop PUT"] [IndexBuildID=432665387088216066] [reload=false]
[2022/05/01 21:29:13.253 +08:00] [ERROR] [node_manager.go:128] ["IndexCoord NodeManager PeekClient"] ["There is no IndexNode client corresponding to NodeID"=-1] [stack="github.com/milvus-io/milvus/internal/indexcoord.(*NodeManager).PeekClient\n\t/go/src/github.com/milvus-io/milvus/internal/indexcoord/node_manager.go:128\ngithub.com/milvus-io/milvus/internal/indexcoord.(*IndexCoord).assignTaskLoop\n\t/go/src/github.com/milvus-io/milvus/internal/indexcoord/index_coord.go:900"]
[2022/05/01 21:29:13.253 +08:00] [WARN] [index_coord.go:902] ["IndexCoord assignmentTasksLoop can not find available IndexNode"]
[2022/05/01 21:29:13.253 +08:00] [DEBUG] [index_coord.go:880] ["IndexCoord assignTaskLoop ctx Done"]
[2022/05/01 21:29:13.257 +08:00] [ERROR] [roles.go:496] ["Get signal to exit\n"] [signal=interrupt] [stack="github.com/milvus-io/milvus/cmd/roles.(*MilvusRoles).Run\n\t/go/src/github.com/milvus-io/milvus/cmd/roles/roles.go:496\nmain.main\n\t/go/src/github.com/milvus-io/milvus/cmd/main.go:296\nruntime.main\n\t/usr/local/go/src/runtime/proc.go:225"]
[2022/05/01 21:29:13.257 +08:00] [DEBUG] [service.go:133] ["Indexcoord stop"] [Address=10.244.9.27:31000]
[2022/05/01 21:29:13.257 +08:00] [INFO] [index_coord.go:299] ["cancel the loop of IndexCoord"]
[2022/05/01 21:29:13.257 +08:00] [INFO] [index_coord.go:304] ["close the task scheduler of IndexCoord"]
[2022/05/01 21:29:13.260 +08:00] [INFO] [grpclog.go:37] ["[core]Subchannel Connectivity change to CONNECTING"]
[2022/05/01 21:29:13.261 +08:00] [INFO] [grpclog.go:37] ["[transport]transport: loopyWriter.run returning. connection error: desc = \"transport is closing\""]
[2022/05/01 21:29:13.261 +08:00] [INFO] [grpclog.go:37] ["[core]Subchannel picks a new address \"10.245.18.8:21121\" to connect"]
[2022/05/01 21:29:13.261 +08:00] [INFO] [grpclog.go:37] ["[core]pickfirstBalancer: UpdateSubConnState: 0xc0005607b0, {CONNECTING <nil>}"]
[2022/05/01 21:29:13.261 +08:00] [INFO] [grpclog.go:37] ["[core]Channel Connectivity change to CONNECTING"]
[2022/05/01 21:29:13.261 +08:00] [WARN] [grpclog.go:46] ["[core]grpc: addrConn.createTransport failed to connect to {10.245.18.8:21121 10.245.18.8:21121 <nil> 0 <nil>}. Err: connection error: desc = \"transport: Error while dialing dial tcp 10.245.18.8:21121: connect: connection refused\". Reconnecting..."]
[2022/05/01 21:29:13.261 +08:00] [INFO] [grpclog.go:37] ["[core]Subchannel Connectivity change to TRANSIENT_FAILURE"]
[2022/05/01 21:29:13.261 +08:00] [INFO] [grpclog.go:37] ["[core]pickfirstBalancer: UpdateSubConnState: 0xc0005607b0, {TRANSIENT_FAILURE connection error: desc = \"transport: Error while dialing dial tcp 10.245.18.8:21121: connect: connection refused\"}"]
[2022/05/01 21:29:13.261 +08:00] [INFO] [grpclog.go:37] ["[core]Channel Connectivity change to TRANSIENT_FAILURE"]
{"level":"warn","ts":"2022-05-01T21:29:13.262+0800","logger":"etcd-client","caller":"v3@v3.5.0/retry_interceptor.go:62","msg":"retrying of unary invoker failed","target":"etcd-endpoints://0xc0006a2c40/#initially=[milvus.etcdc1.corpautohome.com:2384;milvus.etcdc2.corpautohome.com:2384;milvus.etcdc3.corpautohome.com:2384]","attempt":0,"error":"rpc error: code = NotFound desc = etcdserver: requested lease not found"}
[2022/05/01 21:29:13.262 +08:00] [DEBUG] [service.go:147] ["Graceful stop grpc server..."]
[2022/05/01 21:29:13.262 +08:00] [DEBUG] [service.go:240] ["IndexCoord grpcServer loop exit"]
[2022/05/01 21:29:13.262 +08:00] [INFO] [grpclog.go:37] ["[core]Channel Connectivity change to SHUTDOWN"]
[2022/05/01 21:29:13.262 +08:00] [INFO] [grpclog.go:37] ["[core]Subchannel Connectivity change to SHUTDOWN"]
[2022/05/01 21:29:13.262 +08:00] [INFO] [grpclog.go:37] ["[core]Subchannel Connectivity change to SHUTDOWN"]
[2022/05/01 21:29:13.262 +08:00] [INFO] [grpclog.go:37] ["[core]Subchannel Connectivity change to SHUTDOWN"]
[2022/05/01 21:29:13.262 +08:00] [INFO] [grpclog.go:37] ["[transport]transport: loopyWriter.run returning. connection error: desc = \"transport is closing\""]
[2022/05/01 21:29:13.262 +08:00] [INFO] [grpclog.go:37] ["[transport]transport: loopyWriter.run returning. connection error: desc = \"transport is closing\""]
[2022/05/01 21:29:13.262 +08:00] [INFO] [grpclog.go:37] ["[transport]transport: loopyWriter.run returning. connection error: desc = \"transport is closing\""]







indexnode log:
[2022/04/30 15:29:42.260 +08:00] [DEBUG] [task.go:393] ["IndexNode deserialize data success"] [taskID=432701432722096162] [IndexID=432856750151958529] ["index buildID"=432701432722096162] [collectionID=432856676948246529] [partitionID=432856676948246530] [segmentID=432856727266787329]
[2022/04/30 15:29:42.260 +08:00] [DEBUG] [time_recorder.go:78] ["IndexBuildTask 432701432722096162: deserialize vector data done (0ms)"]
[2022/04/30 15:29:42.260 +08:00] [DEBUG] [index.go:219] ["before BuildFloatVecIndexWithoutIds"]
Konwhere BuildWithoutIds config_ is {"SLICE_SIZE":4,"dim":768,"index_type":"IVF_PQ","m":12,"metric_type":"IP","nbits":8,"nlist":2048,"rows":5116}
2022-04-30 15:29:42,272 WARNING [default] [KNOWHERE][MatchNlist][milvus] Row num 5116 match nlist 131
WARNING clustering 5116 points to 256 centroids: please provide at least 9984 training points
WARNING clustering 5116 points to 256 centroids: please provide at least 9984 training points
WARNING clustering 5116 points to 256 centroids: please provide at least 9984 training points
WARNING clustering 5116 points to 256 centroids: please provide at least 9984 training points
WARNING clustering 5116 points to 256 centroids: please provide at least 9984 training points
WARNING clustering 5116 points to 256 centroids: please provide at least 9984 training points
WARNING clustering 5116 points to 256 centroids: please provide at least 9984 training points
WARNING clustering 5116 points to 256 centroids: please provide at least 9984 training points
WARNING clustering 5116 points to 256 centroids: please provide at least 9984 training points
WARNING clustering 5116 points to 256 centroids: please provide at least 9984 training points
WARNING clustering 5116 points to 256 centroids: please provide at least 9984 training points
WARNING clustering 5116 points to 256 centroids: please provide at least 9984 training points
2022-04-30 15:29:54,621 DEBUG [default] [user@milvus-all-indexnode-27603-79df85b494-564zm] [void milvus::knowhere::TimeRecorder::PrintTimeRecord(const string&, double)] [..make_build/3rdparty_download/knowhere-subbuild/src/knowhere_ep/knowhere/knowhere/common/Timer.cpp:49] [KNOWHERE][PrintTimeRecord][milvus] BuildWithoutIds: TrainAndAdd (12.261683 seconds [12261.682596 ms])
2022-04-30 15:29:54,621 DEBUG [default] [user@milvus-all-indexnode-27603-79df85b494-564zm] [void milvus::knowhere::TimeRecorder::PrintTimeRecord(const string&, double)] [..make_build/3rdparty_download/knowhere-subbuild/src/knowhere_ep/knowhere/knowhere/common/Timer.cpp:49] [KNOWHERE][PrintTimeRecord][milvus] BuildWithoutIds: Done (12.316354 seconds [12316.353873 ms])
[2022/04/30 15:29:54.621 +08:00] [DEBUG] [time_recorder.go:78] ["IndexBuildTask 432701432722096162: build index done (0ms)"]
[2022/04/30 15:29:54.657 +08:00] [DEBUG] [time_recorder.go:78] ["IndexBuildTask 432701432722096162: index serialize done (35ms)"]
[2022/04/30 15:29:54.707 +08:00] [DEBUG] [time_recorder.go:78] ["IndexBuildTask 432701432722096162: index codec serialize done (50ms)"]
[2022/04/30 15:29:54.707 +08:00] [DEBUG] [parallel.go:47] [saveIndexFile] [total=2]
[2022/04/30 15:29:54.707 +08:00] [DEBUG] [parallel.go:48] [saveIndexFile] [nPerBatch=1]
[2022/04/30 15:29:54.707 +08:00] [DEBUG] [parallel.go:101] [saveIndexFile] [NumOfGoRoutines=2]
[2022/04/30 15:29:55.026 +08:00] [DEBUG] [parallel.go:43] [saveIndexFile] ["time cost"=319.42146ms]
[2022/04/30 15:29:55.026 +08:00] [DEBUG] [time_recorder.go:78] ["IndexBuildTask 432701432722096162: index file save done (319ms)"]
[2022/04/30 15:29:55.026 +08:00] [DEBUG] [time_recorder.go:78] ["IndexBuildTask 432701432722096162: index building all done (13511ms)"]
[2022/04/30 15:29:55.026 +08:00] [INFO] [task.go:599] ["IndexNode CreateIndex successfully "] [collect=432856676948246529] [partition=432856676948246530] [segment=432856727266787329]
[2022/04/30 15:29:55.036 +08:00] [DEBUG] [task.go:279] ["IndexNode IndexBuildTask PostExecute..."] [buildId=432701432722096162]
[2022/04/30 15:29:55.058 +08:00] [INFO] [task.go:236] ["IndexNode IndexBuildTask saveIndexmeta indexMeta.state to IndexState_Unissued"] [TaskState=Normal] [IndexBuildID=432701432722096162]
[2022/04/30 15:29:55.100 +08:00] [DEBUG] [time_recorder.go:78] ["IndexBuildTask 432701432722096162: IndexNode IndexBuildTask saveIndexMeta (73ms)"]
[2022/05/01 21:29:11.665 +08:00] [WARN] [session_util.go:261] ["session keepalive channel closed"]
[2022/05/01 21:29:11.710 +08:00] [WARN] [session_util.go:441] ["connection lost detected, shuting down"]
[2022/05/01 21:29:11.735 +08:00] [ERROR] [indexnode.go:128] ["Index Node disconnected from etcd, process will exit"] ["Server Id"=276] [stack="github.com/milvus-io/milvus/internal/indexnode.(*IndexNode).Register.func1\n\t/go/src/github.com/milvus-io/milvus/internal/indexnode/indexnode.go:128"]
{"level":"warn","ts":"2022-05-01T21:29:12.780+0800","logger":"etcd-client","caller":"v3@v3.5.0/retry_interceptor.go:62","msg":"retrying of unary invoker failed","target":"etcd-endpoints://0xc000357340/#initially=[milvus.etcdc1.corpautohome.com:2384;milvus.etcdc2.corpautohome.com:2384;milvus.etcdc3.corpautohome.com:2384]","attempt":0,"error":"rpc error: code = Unknown desc = context deadline exceeded"}
[2022/05/01 21:29:12.796 +08:00] [DEBUG] [indexnode.go:244] ["Index node stopped."]
[2022/05/01 21:29:12.830 +08:00] [ERROR] [roles.go:496] ["Get signal to exit\n"] [signal=interrupt] [stack="github.com/milvus-io/milvus/cmd/roles.(*MilvusRoles).Run\n\t/go/src/github.com/milvus-io/milvus/cmd/roles/roles.go:496\nmain.main\n\t/go/src/github.com/milvus-io/milvus/cmd/main.go:296\nruntime.main\n\t/usr/local/go/src/runtime/proc.go:225"]
[2022/05/01 21:29:12.830 +08:00] [DEBUG] [service.go:185] ["IndexNode stop"] [Address=10.245.18.8:21121]
[2022/05/01 21:29:13.230 +08:00] [DEBUG] [indexnode.go:244] ["Index node stopped."]
[2022/05/01 21:29:13.230 +08:00] [DEBUG] [service.go:199] ["Graceful stop grpc server..."]
[2022/05/01 21:29:13.246 +08:00] [INFO] [grpclog.go:37] ["[core]Channel Connectivity change to SHUTDOWN"]
[2022/05/01 21:29:13.246 +08:00] [INFO] [grpclog.go:37] ["[transport]transport: loopyWriter.run returning. connection error: desc = \"transport is closing\""]
[2022/05/01 21:29:13.275 +08:00] [INFO] [grpclog.go:37] ["[core]Subchannel Connectivity change to SHUTDOWN"]
[2022/05/01 21:29:13.275 +08:00] [INFO] [grpclog.go:37] ["[core]Subchannel Connectivity change to SHUTDOWN"]
[2022/05/01 21:29:13.275 +08:00] [INFO] [grpclog.go:37] ["[core]Subchannel Connectivity change to SHUTDOWN"]
[2022/05/01 21:29:13.275 +08:00] [INFO] [grpclog.go:37] ["[transport]transport: loopyWriter.run returning. connection error: desc = \"transport is closing\""]
[2022/05/01 21:29:13.276 +08:00] [INFO] [grpclog.go:37] ["[transport]transport: loopyWriter.run returning. connection error: desc = \"transport is closing\""]
[2022/05/01 21:29:13.276 +08:00] [INFO] [grpclog.go:37] ["[transport]transport: loopyWriter.run returning. connection error: desc = \"transport is closing\""]
```

### Anything else?

N/A

**Comment by xiaofan-luan** [comment]
> ```gfm
> Index Node disconnected from etcd, process will exit
> ```

where did you see memory efficient log?
From the error log:
Index Node disconnected from etcd, process will exit
It definitely point to an etcd issue~

**Comment by JackLCL** [comment]
[2022/05/01 21:29:13.243 +08:00] [ERROR] [node_manager.go:121] ["No IndexNode available"] ["data size"=2397014016] ["IndexNode must have memory size"=14382084096] [stack="github.com/milvus-io/milvus/internal/indexcoord.(*NodeManager).PeekClient\n\t/go/src/github.com/milvus-io/milvus/internal/indexcoord/node_manager.go:121\ngithub.com/milvus-io/milvus/internal/indexcoord.(*IndexCoord).assignTaskLoop\n\t/go/src/github.com/milvus-io/milvus/internal/indexcoord/index_coord.go:900"]
[2022/05/01 21:29:13.243 +08:00] [ERROR] [node_manager.go:128] ["IndexCoord NodeManager PeekClient"] ["There is no IndexNode client corresponding to NodeID"=-1] [stack="github.com/milvus-io/milvus/internal/indexcoord.(*NodeManager).PeekClient\n\t/go/src/github.com/milvus-io/milvus/internal/indexcoord/node_manager.go:128\ngithub.com/milvus-io/milvus/internal/indexcoord.(*IndexCoord).assignTaskLoop\n\t/go/src/github.com/milvus-io/milvus/internal/indexcoord/index_coord.go:900"]
 

From the two lines of logs in indexcoord.
